coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala

```diff
@@ -53,6 +53,8 @@ import filodb.query.exec._
     val earliestRawTime = earliestRawTimestampFn
     val offsetMillis = LogicalPlanUtils.getOffsetMillis(periodicSeriesPlan)
     val (maxOffset, minOffset) = (offsetMillis.max, offsetMillis.min)
+    require(periodicSeriesPlan.validateAtModifier(earliestRawTime),
+      s"$periodicSeriesPlan @modifier and query range should be all greater or less than $earliestRawTime")
 
     val lookbackMs = LogicalPlanUtils.getLookBackMillis(periodicSeriesPlan).max
     val startWithOffsetMs = periodicSeriesPlan.startMs - maxOffset
```

Comment:
Would it be straightforward/correct to use these values to determine whether-or-not the query needs to be rejected? e.g:
```
startWithOffset - lookbackMs, all_at_timestamps >= earliestRawTime -> OK
endWithOffset, all_at_timestamps < earliestRawTime -> OK
else -> REJECT
```

So, the only additional method we'd need is something like `getAllAtTimestampMillis` that just walks the plan and aggregates a list of at-modifier timestamps from the leaves.

(Please sanity-check that this is consistent with the below if-else chain if you do explore this option.)

Reply:
Is there a reason the `getDataSource`/`hasAtModifier` implementation is preferable?

Reply:
not needed.